tests/phpunit/CRM/Upgrade/Incremental/BaseTest.php

```diff
@@ -64,6 +66,33 @@ public function testMessageTemplateStringReplace(): void {
     $this->revertTemplateToReservedTemplate('contribution_invoice_receipt');
   }
 
+  /**
+   * Test that a $this->string replacement in a message template can be done.
+   *
+   * @throws \API_Exception
+   */
+  public function testActionScheduleStringReplace(): void {
+    ActionSchedule::create(FALSE)->setValues([
+      'title' => 'schedule',
+      'absolute_date' => '2021-01-01',
+      'start_action_date' => '2021-01-01',
+      'mapping_id' => 1,
+      'entity_value' => 1,
+      'body_text' => 'blah {contribution.status}',
+      'body_html' => 'blah {contribution.status}',
+      'subject' => 'blah {contribution.status}',
+    ])->execute();
+
+    $upgrader = new CRM_Upgrade_Incremental_MessageTemplates('5.41.0');
+    $upgrader->replaceTokenInActionSchedule('contribution.status', 'contribution.contribution_status_id:name');
```

Comment:
@eileenmcnaughton shouldn't we change reference from :name to label here as well?

Reply:
hmm - well in that case it's testing the functionality - we will wind up with a range of substitutions so it's not the data that matters per se

Reply:
I just figure if anyone was looking this is kind of testing the upgrade step that we are adding so shouldn't we want it to match the upgrade step that we are doing?

Reply:
Hmm - I'm not convinced it's better but it's not worse to switch to label so I've done it